src/gui/project/window.cpp

```diff
@@ -462,54 +449,31 @@ namespace Toolbox::UI {
         std::strncpy(m_rename_buffer, file_name.c_str(), IM_ARRAYSIZE(m_rename_buffer));
     }
 
-    void ProjectViewWindow::actionPasteIntoIndex(const ModelIndex &index, const MimeData &data) {
-        std::optional<std::string> text = data.get_urls();
-        if (!text) {
-            TOOLBOX_ERROR("Mime data wouldn't return uri list");
-            return;
-        }
-
-        std::vector<std::string_view> urls = splitLines(text.value());
-        for (std::string_view src_path_str : urls) {
-            if (src_path_str.starts_with("file:/")) {
-#ifdef TOOLBOX_PLATFORM_LINUX
-                src_path_str = src_path_str.substr(7);
-#elif defined TOOLBOX_PLATFORM_WINDOWS
-                if (src_path_str.starts_with("file:///")) {
-                    src_path_str = src_path_str.substr(8);
-                } else {
-                    src_path_str = src_path_str.substr(7);
-                }
-#endif
-            } else {
-                TOOLBOX_ERROR_V("Can't copy non-local uri {}", src_path_str);
-                continue;
-            }
-
-            fs_path src_path = src_path_str;
-            m_file_system_model->copy(src_path, index, src_path.filename().string());
+    void ProjectViewWindow::actionPasteIntoIndex(const ModelIndex &index,
+                                                 const std::vector<fs_path> &paths) {
+        for (fs_path src_path : paths) {
+            auto src_path_str = src_path.string();
+            m_file_system_model->copy(src_path_str, index,
+                                      fs_path(src_path_str).filename().string());
```

Comment:
Just do `src_path.filename().string()`

Reply:
Fixed in [this commit](https://github.com/JoshuaMKW/JuniorsToolbox/pull/36/commits/8d64c76f6c21f6624c8624669d00839179b4ad1c)